repobuilder/job.go

```diff
@@ -228,54 +261,97 @@ func (j *Job) signFile(fileName, archiveExtension string, overwrite bool) error
 		"--outputs", "sig",
 	}
 
-	grip.AlertWhenf(strings.HasPrefix(archiveExtension, "."),
-		"extension '%s', has a leading dot, which is almost certainly undesirable.", archiveExtension)
-
-	grip.AlertWhenln(overwrite && len(archiveExtension) != 0,
-		"specified overwrite with an archive extension:", archiveExtension,
-		"this is probably an error, (not impacting packages,) but is passed to the client.")
+	grip.AlertWhen(strings.HasPrefix(archiveExtension, "."),
+		message.Fields{
+			"job_id":    j.ID(),
+			"job_scope": j.Scopes(),
+			"repo":      j.Distro.Name,
+			"version":   j.release.String(),
+			"extension": archiveExtension,
+			"message":   "extension has leading dot, which is ususally problem",
+		})
+
+	grip.CriticalWhen(overwrite && len(archiveExtension) != 0,
+		message.Fields{
+			"job_id":    j.ID(),
+			"job_scope": j.Scopes(),
+			"repo":      j.Distro.Name,
+			"version":   j.release.String(),
+			"extension": archiveExtension,
+			"message":   "specified overwrite with an archive extension",
+			"impact":    "no package impact",
+		})
 
 	if overwrite {
-		grip.Noticef("overwriting existing contents of file '%s' while signing it", fileName)
 		args = append(args, "--package-file-suffix", "")
 	} else {
 		// if we're not overwriting the unsigned source file
 		// with the signed file, then we should remove the
 		// signed artifact before. Unclear if this is needed,
 		// the cronjob did this.
-		grip.Warning(os.Remove(fileName + "." + archiveExtension))
+		grip.Warning(message.WrapError(os.Remove(fileName+"."+archiveExtension),
```

Comment:
Is this still fine if archiveExtension is empty?

Reply:
I don't think this logic has changed and it doesn't lead to any problems, I'm not inclined to change it at this point.